controllers/util/tests/tests.go

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright 2019 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tests
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	csiv1 "github.com/IBM/ibm-block-csi-operator/api/v1"
+	clustersyncer "github.com/IBM/ibm-block-csi-operator/controllers/syncer"
+	"github.com/IBM/ibm-block-csi-operator/pkg/config"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	relativeCrPath = "../config/samples/csi.ibm.com_v1_ibmblockcsi_cr.yaml"
+	nodeContainerName = clustersyncer.NodeContainerName
+	controllerContainerName = clustersyncer.ControllerContainerName
+)
+
+type crYamlConfig struct {
```

Comment:
could we use [`IBMBlockCSI` struct](https://github.com/IBM/ibm-block-csi-operator/blob/v1.9.0/api/v1/ibmblockcsi_types.go#L131) instead?

Reply:
is there an error? what is the issue?

Reply:
there is no error, but the way that the [yaml.v2](https://pkg.go.dev/gopkg.in/yaml.v2@v2.4.0) module is needing the specific path, so for example it can't take `apiVersion`, `kind`, `namespace`, so you need to write them and if you want to convert the [spec](https://github.com/IBM/ibm-block-csi-operator/blob/d0562de677077fd3e9dec3d484cce8140da961c0/controllers/util/tests/tests.go#L44) to [IBMBlockCSISpec](https://github.com/IBM/ibm-block-csi-operator/blob/85cc3c31e3e82aa9dfdcdec02d971f9a19a316e3/api/v1/ibmblockcsi_types.go#L51) you will have code dup on [this](https://github.com/IBM/ibm-block-csi-operator/blob/d0562de677077fd3e9dec3d484cce8140da961c0/controllers/util/tests/tests.go#L124) function

Reply:
done